modules/Bio/EnsEMBL/Compara/RunnableDB/GeneTrees/DumpWGAExpectedTags.pm

```diff
@@ -0,0 +1,71 @@
+=head1 LICENSE
+
+See the NOTICE file distributed with this work for additional information
+regarding copyright ownership.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=head1 NAME
+
+Bio::EnsEMBL::Compara::RunnableDB::GeneTrees::DumpWGAExpectedTags
+
+=head1 DESCRIPTION
+
+A runnable to dump wga_expected tag and delete it from a database.
+
+=cut
+
+package Bio::EnsEMBL::Compara::RunnableDB::GeneTrees::DumpWGAExpectedTags;
+
+use strict;
+use warnings;
+
+use base ('Bio::EnsEMBL::Compara::RunnableDB::BaseRunnable');
+
+
+sub run {
+    my ($self) = shift @_;
+   
+    $self->dump_wga_expected();
+
+    $self->delete_wga_expected();
+}
+
+
+sub dump_wga_expected {
+    my ($self) = shift @_;
```

Comment:
This is quite an unusual way to get the arguments. If you look at examples throughout the codebase, we generally do this sort of thing in one of two ways:
`my $self = shift;` : for cases where we're **not** expecting any additional arguments, or
`my ($self, $x, $y) = @_;` : for cases where we _are_ expecting additional arguments.

In perl, `$_` and `@_` are like defaults, so calling `shift` is the same as calling `shift @_`. But you're also kind of dereferencing the first element of `@_` here by putting `($self)`. It works in this case because perl does a lot of guesswork on your behalf (which is bad sometimes because it masks things like this) and happened to guess what you're trying to do correctly in this case, but in a more complex case, I think this could lead to issues. Would recommend:
```suggestion
    my $self = shift;
```

Reply:
Thank you very much for a concise explanation @carlacummins :)